core/Pos/Crypto/Signing.hs

```diff
@@ -247,27 +247,24 @@ verifyProxySecretKey ProxySecretKey{..} =
 
 -- | Delegate signature made with certificate-based permission. @a@
 -- stays for message type used in proxy (ω in the implementation
--- notes), @b@ for type of message signed.
+-- notes), @b@ for type of message signed. We add whole psk as a field
+-- because otherwise we can't verify sig in heavyweight psk transitive
+-- delegation: i -> x -> d, we have psk from x to d, slot leader is
+-- i.
 data ProxySignature w a = ProxySignature
-    { pdOmega      :: w
-    , pdDelegatePk :: PublicKey
-    , pdCert       :: ProxyCert w
-    , pdSig        :: CC.XSignature
+    { pdPsk :: ProxySecretKey w
+    , pdSig :: CC.XSignature
```

Comment:
Lol, no idea. Proxy Delegation? Got used to it. Will fix in https://issues.serokell.io/issue/CSL-1171